odl/trafos/non_uniform_fourier.py

```diff
@@ -9,123 +9,242 @@
+            It must have shape ``(M, d)``, where ``M`` is the number of
+            samples and ``d`` the dimension of ``domain``. The magnitude
+            of the samples may not exceed ``pi / s``, where ``s`` is
+            ``domain.cell_sides``.
+        range : `TensorSpace`, optional
+            Range of the non-uniform FFT operator. It must be a
+            one-dimensional space with shape ``(M,)``. By default, it is
+            chosen as ``cn(M)`` with the same arithmetic precision as
+            ``domain.dtype``.
         skip_normalization : bool, optional
-            Whether the normalization step should be skipped
+            Whether the samples normalization step should be skipped.
+            If ``True``, all ``samples`` are expected to lie between -0.5
+            (inclusive) and +0.5 (exclusive).
+        kwargs
+            Additional keyword arguments passed on to the ``NFFT`` class
+            constructor, except for its ``N`` and ``M`` parameters.
+
+        Notes
+        -----
+        - The ``pynfft`` backend library currently only supports double
+          precision arithmetic, thus all computations are carried out with
+          double precision irrespective of the ``dtype`` of ``domain`` and
+          ``range``.
         """
-        if not isinstance(space, DiscreteLp) or not space.is_uniform:
-            raise ValueError("`space` should be a uniform `DiscreteLp`")
         super(NonUniformFourierTransform, self).__init__(
-            space=space,
-            samples=samples,
-            domain=space,
-            range=cn(len(samples)),
-            skip_normalization=skip_normalization,
-        )
-
-    @property
-    def adjoint(self):
-        return NonUniformFourierTransformAdjoint(
-            shape=self.shape,
-            samples=self.samples,
-            skip_normalization=True,
+            domain, samples, range, **kwargs
         )
 
     def _call(self, x):
-        """Compute the direct non uniform FFT.
-
-        Parameters
-        ----------
-        x : `numpy.ndarray`
-            The data whose non uniform FFT you want to compute
-
-        Returns
-        -------
-        out_normalized : `numpy.ndarray`
-            Result of the transform
-        """
+        """Return ``self(x)``."""
         if not self._has_run:
-            self._normalize()
+            if not self._skip_normalization:
+                normalize_samples(
+                    self.samples, self.domain.cell_sides, out=self.samples
+                )
             self.nfft.precompute()
+            self.nfft.x = self.samples
             self._has_run = True
+
         self.nfft.f_hat = np.asarray(x)
         out = self.nfft.trafo()
-        # The normalization is inspired from
-        # https://github.com/CEA-COSMIC/pysap-mri/blob/master/mri/reconstruct/fourier.py#L123
-        out /= np.sqrt(self.nfft.M)
+        out *= self.domain.cell_volume / (2 * np.pi)
```

Comment:
Oh and actually I don't understand this as well !

Reply:
This one is because ODL transforms try to approximate integrals. This is also detailed in the [Fourier trafo documentation](https://odlgroup.github.io/odl/math/trafos/fourier_transform.html).

Reply:
I guess what I am more concerned about is the fact that the normalization is not the same for the adjoint, potentially (if `range.cell_volume` and `domain.cell_volume` differ). This will mean that the adjoint does not verify the adjoint property anymore (`<x, Ty> = <T*x, y>`).

Reply:
Absolutely. I have so far only looked into the forward transform, the adjoint needs a different scaling as you point out.

Reply:
Ok makes sense then! Basically the 2 scalings would need to be the same.